Point2D coords
float32[4] covariance
uint8 color